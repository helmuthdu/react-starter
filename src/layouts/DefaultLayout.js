// @flow
import React, { Fragment, PureComponent } from 'react';
import { Route } from 'react-router-dom';

interface DefaultLayoutProps {
  children: any;
}

interface DefaultLayoutState {}

export default class DefaultLayout extends PureComponent<DefaultLayoutProps, DefaultLayoutState> {
  render() {
    // const { pathname } = this.props.children.props.location;

    return (
      <Fragment>
        <div>{this.props.children}</div>
      </Fragment>
    );
  }
}

export const DefaultLayoutRoute = ({ component: Component, ...rest }: any) => {
  return (
    <Route
      {...rest}
      render={(matchProps: DefaultLayoutProps) => (
        <DefaultLayout>
          <Component {...matchProps} />
        </DefaultLayout>
      )}
    />
  );
};
