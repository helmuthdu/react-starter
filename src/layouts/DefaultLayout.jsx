// @flow
import React, { Fragment } from 'react';
import { Route } from 'react-router-dom';
import { createGlobalStyle } from 'styled-components';

const GlobalStyles = createGlobalStyle`
  @import url('https://fonts.googleapis.com/css?family=Montserrat|Source+Sans+Pro');

  body,
  html {
    overflow-y: auto;
    overflow-x: hidden;
    margin: 0;
    padding: 0;
    font-family: 'Source Sans Pro', sans-serif;
    background-color: #f1f1f1;
    font-size: 16px;
  }

  h1, h2, h3, h4, h5, h6 {
    margin: 0 0 1rem;
  }
`;

export const DefaultLayout = ({ children }: any) => {
  return <Fragment>{children}</Fragment>;
};

export const DefaultLayoutRoute = ({ component: Component, ...rest }: any) => {
  return (
    <Route
      {...rest}
      render={(matchProps: any) => (
        <DefaultLayout>
          <GlobalStyles />
          <Component {...matchProps} />
        </DefaultLayout>
      )}
    />
  );
};

export default DefaultLayoutRoute;
