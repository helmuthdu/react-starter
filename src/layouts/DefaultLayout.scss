@import url('https://fonts.googleapis.com/css?family=Montserrat|Source+Sans+Pro');

body,
html {
  overflow-y: auto;
  overflow-x: hidden;
  margin: 0;
  padding: 0;
  font-family: 'Source Sans Pro', sans-serif;
  background-color: #f1f1f1;
  font-size: 16px;
}

h1,
h2,
h3,
h4,
h5,
h6 {
  margin: 0 0 1rem;
}
