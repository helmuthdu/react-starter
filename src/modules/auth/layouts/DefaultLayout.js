// @flow
import React, { Fragment } from 'react';
import { Route } from 'react-router-dom';
import { injectGlobal } from 'styled-components';

injectGlobal`
  @import url('https://fonts.googleapis.com/css?family=Montserrat|Source+Sans+Pro');

  body,
  html {
    overflow-y: auto;
    overflow-x: hidden;
    margin: 0;
    padding: 0;
    font-family: 'Source Sans Pro';
    background-color: #f1f1f1;
    font-size: 16px;
  }

  h1, h2, h3, h4, h5, h6 {
    margin: 0 0 1rem;
  }
`;

interface DefaultLayoutProps {
  children: any;
}

export const DefaultLayout = ({ children }: DefaultLayoutProps) => {
  return (
    <Fragment>
      <div>{children}</div>
    </Fragment>
  );
};

export const DefaultLayoutRoute = ({ component: Component, ...rest }: any) => {
  return (
    <Route
      {...rest}
      render={(matchProps: DefaultLayoutProps) => (
        <DefaultLayout>
          <Component {...matchProps} />
        </DefaultLayout>
      )}
    />
  );
};

export default DefaultLayoutRoute;
