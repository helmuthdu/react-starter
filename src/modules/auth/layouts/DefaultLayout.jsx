// @flow
import React, { Fragment } from 'react';
import { Route } from 'react-router-dom';
import { createGlobalStyle } from 'styled-components';

const GlobalStyles = createGlobalStyle`
  @import url('https://fonts.googleapis.com/css?family=Montserrat|Source+Sans+Pro');

  body,
  html {
    overflow-y: auto;
    overflow-x: hidden;
    margin: 0;
    padding: 0;
    font-family: 'Source Sans Pro', sans-serif;
    background-color: #ffffff;
    font-size: 16px;
  }
`;

export const DefaultLayout = ({ children }: any) => {
  return <Fragment>{children}</Fragment>;
};

export const DefaultLayoutRoute = ({ component: Component, ...rest }: any) => {
  return (
    <Route
      {...rest}
      render={(matchProps: any) => (
        <DefaultLayout>
          <GlobalStyles />
          <Component {...matchProps} />
        </DefaultLayout>
      )}
    />
  );
};

export default DefaultLayoutRoute;
