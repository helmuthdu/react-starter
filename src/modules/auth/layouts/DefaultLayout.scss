@import url('https://fonts.googleapis.com/css?family=Montserrat|Source+Sans+Pro');

body,
html {
  overflow-y: auto;
  overflow-x: hidden;
  margin: 0;
  padding: 0;
  font-family: 'Source Sans Pro', sans-serif;
  background-color: #ffffff;
  font-size: 16px;
}
