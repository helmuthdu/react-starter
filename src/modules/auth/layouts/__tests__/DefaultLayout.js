import { shallow } from 'enzyme';
import React from 'react';
import { DefaultLayout, DefaultLayoutRoute } from '../DefaultLayout';

describe('auth/layouts -> DefaultLayout component', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(
      <DefaultLayout>
        <h1>Content</h1>
      </DefaultLayout>
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('auth/layouts -> DefaultLayoutRoute component', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<DefaultLayoutRoute />);
    expect(wrapper).toMatchSnapshot();
  });
});
