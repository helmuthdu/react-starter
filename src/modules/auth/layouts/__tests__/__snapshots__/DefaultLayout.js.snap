// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth/layouts -> DefaultLayout component renders without crashing 1`] = `
<Fragment>
  <h1>
    Content
  </h1>
</Fragment>
`;

exports[`auth/layouts -> DefaultLayoutRoute component renders without crashing 1`] = `
<Route
  render={[Function]}
/>
`;
